Automattic/wp-calypso
# Checkout: Modify applePayProcessor to use cart for creating transaction

#### Changes proposed in this Pull Request

This PR modifies the Apple Pay payment processor to use the shopping cart for creating the transaction request cart, rather than converting the line items, since the line items themselves have been converted from the shopping cart. This is the same refactoring that was done for other processors in #51336, #51240, #51460, and #51156. As part of this work, this also converts the processor to TypeScript.

This is the last processor that needs to be converted.

This also adds explicit error handling during the transaction so that #51427 will be able to trigger error boundaries for explicit errors.

#### Testing instructions

- Complete a purchase using Apple Pay as the payment method (instructions to make Apple Pay work in a test environment are complex; ask me if you have a question).
- Verify that the purchase completes successfully.
